test/fixtures/es-module-loaders/hook-resolve-type.mjs

```diff
@@ -1,6 +1,29 @@
 let importedESM = 0;
 let importedCJS = 0;
-global.getModuleTypeStats = () => { return {importedESM, importedCJS} };
+
+export function globalPreload({ port }) {
+  port.on('message', (int32) => {
+    port.postMessage({ importedESM, importedCJS });
+    Atomics.store(int32, 0, 1);
+    Atomics.notify(int32, 0);
+  });
+  port.unref();
+  return `
+    const { receiveMessageOnPort } = getBuiltin('worker_threads');
+    global.getModuleTypeStats = async function getModuleTypeStats() {
+      const sab = new SharedArrayBuffer(4);
+      const int32 = new Int32Array(sab);
+      port.postMessage(int32);
+      // Artificial timeout to keep the event loop alive.
+      // https://bugs.chromium.org/p/v8/issues/detail?id=13238
+      // TODO(targos) Remove when V8 issue is resolved.
+      const timeout = setTimeout(() => {}, 1_000);
+      await Atomics.waitAsync(int32, 0, 0).value;
+      clearTimeout(timeout);
```

Comment:
/cc @joyeecheung this is related to https://bugs.chromium.org/p/v8/issues/detail?id=13238 / https://github.com/nodejs/node/pull/44409

Reply:
FWIW the current implementation of #44710 relies on `Atomics.waitAsync` not keeping the event loop alive.

Reply:
@aduh95 What would be your expectation? Should `Atomics.waitAsync` keep the event loop alive?

Reply:
Ideally there would be a way to control it, like you can opt-out for `setTimeout`. My expectation was that it would not keep the event loop alive, but I could see how that can also be annoying for some (most?) use cases. Does it also not keep the event loop alive if you add a fourth timeout parameter?

Reply:
The fourth timeout parameter doesn't change the behavior.

Reply:
I think so far the vibes I got is that "this is host-defined" i.e. Node.js gets to decide if the timeout parameter keeps the event loop alive, but also if we want to implement that, V8 needs to provide a way in the platform API for canceling a delayed task, which is probably too much complexity, so to be lazy we can also just keep the things the way they are (not let the time out keep the event loop alive, which some may argue is also their expected behavior).